ProcessSmarterTestPackage/PostProcessors/FormPartitionPostProcessor.cs

```diff
@@ -16,23 +16,101 @@ public override IList<ValidationError> GenerateErrors()
 
             var partitionIdentifier = Processor.ChildNodeWithName("identifier").ValueForAttribute("uniqueid");
             var itemGroups = Processor.ChildNodesWithName("itemgroup").ToList();
+            const string groupIdPattern = @"(\d+-\d+):([G|I])-(\d+-\d+)";
+
             foreach (var itemGroup in itemGroups)
             {
-                if (!itemGroup.ChildNodeWithName("identifier")
-                    .ValueForAttribute("uniqueid").Split(':').First()
-                    .Equals(partitionIdentifier, StringComparison.OrdinalIgnoreCase))
+                var match = Regex.Match(itemGroup.ChildNodeWithName("identifier")
+                    .ValueForAttribute("uniqueid"), groupIdPattern);
+                if (!match.Success)
                 {
                     result.Add(new ValidationError
                     {
-                        ErrorSeverity = ErrorSeverity.Benign,
+                        ErrorSeverity = ErrorSeverity.Severe,
```

Comment:
These identifiers are parsed when retrieving items to present in Student. If they do not follow this pattern, they may cause (hard to debug) failures at runtime.